impl/client/src/main/javascript/web/pentaho/type/action/ListChangeset.js

```diff
@@ -125,12 +125,78 @@ define([
      *
      * @param {string} key - The key of the element.
      *
-     * @return {pentaho.type.action.ComplexChangeset} The child changeset or `null`.
+     * @return {pentaho.type.action.Changeset} The child changeset or `null`.
      */
     getChange: function(key) {
       return O.getOwn(this.__changesetByKey, key) || null;
     },
 
+
+    /** @inheritDoc */
+    compose: function(changeset) {
+
+      if(!(changeset instanceof ListChangeset)) {
+        throw error.argInvalidType("changeset", "pentaho.type.action.ListChangeset", typeof changeset);
+      }
+
+      if(changeset === this) {
+        throw error.argInvalid("changeset", "Changeset is equal to itself");
+      }
+
+      var target = changeset.target;
+      if(target !== this.target) {
+        throw error.operInvalid("Can not compose changeset that has a different target");
```

Comment:
"argInvalid"
"Cannot compose with a changeset which has a different target" (and change this in the complex version as well, please.